plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -345,6 +350,36 @@ func (s *SQLServer) Stop() {
 	}
 }
 
+func (s *SQLServer) gatherServerWithRetry(pool *sql.DB, query Query, acc telegraf.Accumulator, connectionString string) error {
+	for retryAttempt := 0; retryAttempt <= s.RetryCount; retryAttempt++ {
+		queryError := s.gatherServer(pool, query, acc, connectionString)
+		if queryError == nil {
+			// gatherServer completed successfully
+			break
+		}
+
+		if retryAttempt >= s.RetryCount {
+			// gatherServer failed on the final retry attempt
+			return queryError
+		}
+
+		log.Printf("W! [inputs.sqlserver] Error gathering data, retrying %d more time(s): %s.", s.RetryCount-retryAttempt, queryError)
+
+		// Calculate the wait time - either constant or exponential
+		multiplier := 1.0
+		if s.RetryExponentialBackoff {
+			multiplier = math.Pow(2, float64(retryAttempt))
+		}
```

Comment:
If you initialize `multiplier` _before_ the retry loop we do not need `math` by doing
```suggestion
		if s.RetryExponentialBackoff {
			multiplier *= 2
		}
```